### PR TITLE
[node v6] [mapnik v3] Add support to Node v6 and Mapnik v3

### PR DESCRIPTION
**Pull request by dgaubert**
- Based on the latest version from Mapbox's upstream (v0.6.18)
- Targets to mapnik@3.4.15
- Supports Node v6 LTS
- Applies CARTO customizations.

**Note**: supersedes #4

### CODE REVIEW

**Comment by rafatower** [comment]
My understanding is that this PR is `v0.6.18` plus the changes in `cdb-0.x` backported (merged), plus some specific changes to support node v6. I'm taking a look at everything, but I'll focus the review on the later changes.

File: lib/render.js [hunk 1]
```diff
@@ -134,7 +134,13 @@ function getImage(source, image, options, x, y, callback) {
...
                 }
                 return callback(null, buffer);
-            });
+            }
+
+            if (options.format == 'utf') {
+                view.encode(options, encodeCallback);
+            } else {
+                view.encode(options.format, options, encodeCallback);
+            }
```
Comment (rafatower) [inline lib/render.js:143]: Note to myself: the commit 7c947e5598f8e0a476230f5ec2ad92f0487b7bea is easier to understand. Actually on top of `v0.6.18`.

File: Makefile [hunk 2]
```diff
@@ -15,4 +15,10 @@ endif
 check:
 	npm test
 
+testpack:
```
Comment (rafatower) [inline Makefile:18]: Note to myself: from upstream v0.6.18

File: package.json [hunk 3]
```diff
@@ -16,14 +16,15 @@
...
+        "Blake Thompson <flippmoke>"
     ],
     "dependencies": {
         "step"        : "~0.0.5",
-        "generic-pool": "~2.1.1",
-        "mapnik":       "https://s3.amazonaws.com/cartodb-node-binary/mapnik/v1.4.15-cdb10/1.4.15-cdb10.tgz",
-        "mime"        : "~1.2.11",
-        "sphericalmercator": "~1.0.2"
+        "generic-pool": "~2.4.0",
+        "mapnik":       "3.5.14",
```
Comment (rafatower) [inline package.json:25]: are we using the upstream mapnik? yay!

**Review by rafatower — APPROVED** [review]
:+1:

File: package.json (hunk 3 above)
Comment (dgaubert) [reply package.json:25]: yep, we are ;)

**Comment by dgaubert** [comment]
Thanks a lot @rafatower for reviewing this PR. It was the hardest one in the migration.

**Pull request merged**